mesheryctl/internal/cli/root/system/check.go

```diff
@@ -0,0 +1,94 @@
+package system
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"strings"
+
+	meshkitkube "github.com/layer5io/meshkit/utils/kubernetes"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/spf13/cobra"
+)
+
+var PreCheckCmd = &cobra.Command{
+	Use:   "preflight",
+	Short: "run basic checks to verify environment for deployment",
+	Long:  `To verify environment readiness for a Meshery deloyment `,
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runPreDeploymentChecks()
+	},
+}
+
+func runPreDeploymentChecks() error {
+	log.Info("\nDocker \n--------------")
+	//Check whether docker daemon is running or not
+	if err := exec.Command("docker", "ps").Run(); err != nil {
+		log.Error("Docker is not running")
+	}
+	log.Info("√ Docker is running")
+
+	//Check for installed docker-compose on client system
+	if err := exec.Command("docker-compose", "-v").Run(); err != nil {
+		log.Info("Docker-Compose is not available")
+	}
+	log.Info("√ docker-compose is available")
```

Comment:
Docker is not necessary for all types of deployments. This could probably be shown as a warning.

// @leecalcote

Reply:
@piyushsingariya,
Should we have Docker prereqs and K8s prereqs as mutually exclusive ? If either prereq is met, we are good to go.
Can we pass a flag to the preflight to check the prereqs for a certain platform ?
i.e.

`mesheryctl exp preflight docker` checks for below prereqs:

Docker 
--------------
√ Docker is running
√ docker-compose is available

√ √ Meshery prerequisites met

AND

`mesheryctl exp preflight kubernetes` checks for below prereqs:

Kubernetes API
--------------
√ can initialize the client
√ can query the Kubernetes API
√ has required level of privileges

Kubernetes Version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

√ √ Meshery prerequisites met

// @navendu-pottekkat @leecalcote

Reply:
Instead of a flag, we can just check the user's `current-context` and decide right?

Reply:
@navendu-pottekkat should I make it `context` dependent?

Reply:
Let's get some more opinions here-

@leecalcote @hexxdump @metonymic-smokey